src/structures/ApplicationCommand.js

```diff
@@ -0,0 +1,102 @@
+'use strict';
+
+const Base = require('./Base');
+const { ApplicationCommandOptionType } = require('../util/Constants');
+const Snowflake = require('../util/Snowflake');
+
+/**
+ * Represents an application command, see {@link InteractionClient}.
+ * @extends {Base}
+ */
+class ApplicationCommand extends Base {
+  constructor(client, data, guildID) {
+    super(client);
+
+    /**
+     * The ID of the guild this command is part of, if any.
+     * @type {Snowflake?}
+     * @readonly
+     */
+    this.guildID = guildID || null;
+
+    this._patch(data);
+  }
+
+  _patch(data) {
+    /**
+     * The ID of this command.
+     * @type {Snowflake}
+     * @readonly
+     */
+    this.id = data.id;
+
+    /**
+     * The ID of the application which owns this command.
+     * @type {Snowflake}
+     * @readonly
+     */
+    this.appplicationID = data.application_id;
+
+    /**
+     * The name of this command.
+     * @type {string}
+     * @readonly
+     */
+    this.name = data.name;
+
+    /**
+     * The description of this command.
+     * @type {string}
+     * @readonly
+     */
+    this.description = data.description;
+
+    /**
+     * The options of this command.
+     * @type {Object[]}
+     * @readonly
+     */
+    this.options = data.options.map(function m(o) {
+      return {
+        type: ApplicationCommandOptionType[o.type],
+        name: o.name,
+        description: o.description,
+        default: o.default,
+        required: o.required,
+        choices: o.choices,
+        options: o.options ? o.options.map(m) : undefined,
```

Comment:
```suggestion
        options: o.options?.map(m),
```
make use of optional chaining